src/Domain/System/LogGateway.php

```diff
@@ -74,6 +74,37 @@ public function queryLogs(QueryCriteria $criteria, $gibbonSchoolYearID)
                     ->where('gibbonLog.gibbonPersonID = :gibbonPersonID')
                     ->bindValue('gibbonPersonID', $gibbonPersonID);
             },
+            'module' => function ($query, $module) {
+                return $query
+                    ->where('gibbonModule.name = :module')
+                    ->bindValue('module', $module);
+            },
+            'startDate' => function ($query, $startDate) {
+                return $query
+                    ->where('timestamp >= :startDate')
+                    ->bindValue('startDate', $startDate);
+            },
+            'endDate' => function ($query, $endDate) {
+                return $query
+                    ->where('timestamp <= :endDate')
+                    ->bindValue('endDate', $endDate);
+            },
+            'array' => function ($query, $array) {
+                $array = unserialize($array);
+                if (is_array($array)) {
+                    $count = 0;
+                    foreach ($array as $key => $value) {
+                        $bindKey = 'key' . $count;
+                        $bindValue = 'value' . $count;
+
+                        $query
+                            ->where('serialisedArray LIKE CONCAT("%", :'.$bindKey.', "%;%", :'.$bindValue.', "%")')
+                            ->bindValue($bindKey, $key)
+                            ->bindValue($bindValue, $value);
+                    }
```

Comment:
@raynichc: If I read it correctly, you planed to increment the `$count` variable in the foreach loop, right? It seems to miss a `$count++` in your loop.

Reply:
Good catch, I think I accidentally dropped it when I copied the code. I'll fix that up. Thanks!